rfcs/0018-glides-trainsheet-events.md

```diff
@@ -243,226 +264,280 @@ Producers SHOULD continue to produce versions of the event while there are still
           "startLocation": {"gtfsId": "place-lake"},
+          "endLocation": {"gtfsId": "place-gover"},
           "startTime": "10:00:00",
-          "nextTripKey": {
-            "serviceDate": "2022-01-20",
-            "id": "ADDED-2"
-          },
-          "consist": [
+          "endTime": "10:47:00"
+        },
+        "dropped": true,
+        "droppedReason": {"text": "ran as single"},
+        "scheduled": {
+          "cars": [
             {
-              "label": "3850"
-            }
-          ],
-          "operators": [
+              "run": "506",
+              "operator": "678",
+            },
             {
-              "badgeNumber": "567"
+              "run": "507",
+              "operator": "789",
             }
-          ],
-          "comment": "single"
+          ]
         }
-      },
+      }
+    ]
+  }
+}
+{
+  "type": "com.mbta.ctd.glides.trips_updated.v1",
+  "data": {
+    /* author elided */
+    "inputType": "edit-trip",
+    "updatedTrips": [
       {
-        "type": "com.mbta.ctd.glides.trip_added.v1",
-        "data": {
-          "tripKey": {
-            "serviceDate": "2022-01-20",
-            "id": "ADDED-2"
-          },
-          "endLocation": {"gtfsId": "place-lake"},
-          "previousTripKey": {
-            "serviceDate": "2022-01-20",
-            "id": "ADDED-1"
-          },
-          "consist": [
+        "type": "updated",
+        "tripKey": {
+          "serviceDate": "2022-01-20",
+          "startLocation": {"gtfsId": "place-lake"},
+          "endLocation": {"gtfsId": "place-gover"},
+          "startTime": "09:55:00",
+          "endTime": "10:42:00"
+        },
+        "comment": "single",
+        "cars": [
+          {
+            "carNumber": "3800",
+            "operator": "456"
+          }
+          /* note that there is no second car, this is how the second car is removed */
+        ],
+        "scheduled": {
+          "cars": [
             {
-              "label": "3850"
-            }
-          ],
-          "operators": [
+              "run": "504",
+              "operator": "456",
+            },
             {
-              "badgeNumber": "567"
+              "run": "505",
+              "operator": "567",
             }
-          ],
-          "comment": "single"
+          ]
         }
       }
     ]
   }
 }
+{
+  "type": "com.mbta.ctd.glides.trips_updated.v1",
+  "data": {
+    /* author elided */
+    "inputType": "add-trip",
+    "updatedTrips": [
+      {
+        "type": "added",
+        "tripKey": {
+          "serviceDate": "2022-01-20",
+          "id": "ADDED-1"
+        },
+        "startLocation": {"gtfsId": "place-lake"},
+        "startTime": "10:00:00",
+        "nextTripKey": {
+          "serviceDate": "2022-01-20",
+          "id": "ADDED-2"
+        },
+        "cars": [
+          {
+            "carNumber": "3850",
+            "operator": "567"
+          }
+        ],
+        "scheduled": null
```

Comment:
The `scheduled` field isn't optional right now. It could be, but I figured a required `null|object` field would better show that the schedule data will be part of every update to scheduled trips (instead of just the first). An optional field could imply it's missing for added trips only, or could imply that it's missing on some scheduled trips.

That's easy to solve by adding a `MUST` sentence, so if you think an optional field is better than a required nullable field I can change it.

Reply:
I don't have strong feelings either way.